### PR TITLE
impl: Implement Add Product to Cart Function

### PR DESCRIPTION
**Pull request by SuperRedTrain**
In the backend folder:
- Modified productDAO.js in the dao folder and added findProductById function which retrive a product's information from database by id.
- Modified product.controller.js in the api folder and added getProductById function which receive request from the backend, get a product's information by invoking findProductById function, and response to the backend with the product's information.
- Modified shop.route.js in the api folder and added a route which refers to getProductById function.

In the frontend/src folder:
- Added cartConstants.js in the constants folder to define an action type for the cart state.
- Added cartReducers.js in the reducers folder to take an action on the cart state.
- Added ProductScreen.js in the screens folder to create a simple product page, retrive a product's information, and add an Add-to-Cart button on the product page.
- Modified CartScreen.js in the screens folder to fetch the products' information from the cart state.
- Added cartAction.js in the services folder to store a product's information to the cart state and localStorage.
- Added productAction.js in the services folder to get a product's information from the backend by id.
- Modified App.js to add a route which refers to ProductScreen.
- Modified store.js to get cartItems from localStorage and initial the cart state with data stored in localStorage.

Related to Github Issue #58

### CODE REVIEW

**Review by kindubadhan — APPROVED** [review]
It looks good to me.

File: backend/api/product.controller.js [hunk 1]
```diff
@@ -78,4 +78,28 @@ export default class ProductController {
...
+        try {
+            const productId = req.params.id
+
+            console.log(productId)
+
+            let product = await ProductDAO.findProductById(productId)
+
+            if (product) {
+/*                 res.json({
+                    _id: product._id,
```
Comment (London-Alex) [inline backend/api/product.controller.js:93]: I recommend deleting this portion of code that is commented out.

**Review by London-Alex — CHANGES_REQUESTED** [review]
The pull request is appropriately labelled and has the issue linked. The top level description/comment is very detailed. 
I have checked out the branch and tested the functionality on my local machine. It seems to work correctly.
The code looks good and well structured. I will approve the pull request once the small recommendation regarding the commented code is fulfilled.

File: backend/api/product.controller.js (hunk 1 above)
Comment (SuperRedTrain) [reply backend/api/product.controller.js]: Thank you for your recommendation. I have deleted this commented code.

**Review by London-Alex — APPROVED** [review]
Thank you for making that update! Everything looks in order now.

**Pull request merged**